core/src/main/java/me/aurium/beetle/api/task/futures/TaskFuture.java

```diff
@@ -228,6 +230,38 @@ public Executor defaultExecutor() {
         return (TaskFuture<T>) super.exceptionally(fn);
     }
 
+    /**
+     * This does exactly what you think it does. If you have a main thread, don't call this. (Although if you have a main thread,
+     * you should be using {@link SyncTaskerFuture} instead.
+     *
+     * @param unit time unit
+     * @param amount amount of time units to sleep for
+     * @return itself
+     */
+    public TaskFuture<T> thenSleep(TimeUnit unit, long amount) {
+
+        LockSupport.parkUntil(unit.toNanos(amount));
+
+        return this.thenApply(s -> s); //is there a point to this or should i just return ""this""
```

Comment:
There's no point to this. This method shouldn't even return a future - returning a future is very misleading because it implies the operation does not block the calling thread.

Reply:
This should run on the delayed executor then, but i have no idea how to make it run on the delayed executor on the same thread as it is currently on.

Reply:
I think what i'm gonna do is return a thenRun with some sort of sleep, that way it doesn't block?

Reply:
I'm still not sure i understand the logic of the then(something) methods - I understand the then(something)async methods run their logic on the executor provided but i'm unsure of what then(something) methods do.

My current thinking is that if then(something)async methods like thenRunAsync run on another executor, then(something) methods just run on the same executor, but when I asked you this i got a response that is probably more correct but didn't do much to explain to my monkey brain how this actually functioned - and it is key that i understand how this functions to stop making these mistakes haha